Source/Components/ImageGlass.Services/ExplorerSortOrder.cs

```diff
@@ -0,0 +1,104 @@
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+Author: Kevin Routley (aka fire-eggs)
+*/
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Text;
+using ImageGlass.Services.Configuration;
+
+namespace ImageGlass.Services
+{
+    public static class ExplorerSortOrder
+    {
+        // Convert the Explorer column name to our currently available sorting order
+        private static readonly Dictionary<string, ImageOrderBy> SortTranslation = new Dictionary<string, ImageOrderBy>()
+        {
+            { "System.DateModified", ImageOrderBy.LastWriteTime },
+            { "System.ItemDate", ImageOrderBy.LastWriteTime },
+            { "System.ItemTypeText", ImageOrderBy.Extension},
+            { "System.FileExtension", ImageOrderBy.Extension},
+            { "System.FileName", ImageOrderBy.Name},
+            { "System.ItemNameDisplay", ImageOrderBy.Name},
+            { "System.Size", ImageOrderBy.Length },
+            { "System.DateCreated",ImageOrderBy.CreationTime},
+            { "System.DateAccessed",ImageOrderBy.LastAccessTime},
+        };
+
+        [DllImport("ExplorerSortOrder32.dll", CallingConvention = CallingConvention.Cdecl, CharSet = CharSet.Unicode, EntryPoint = "GetExplorerSortOrder")]
+        public static extern int GetExplorerSortOrder32(string path, ref StringBuilder str, int len, ref Int32 ascend);
+
+        [DllImport("ExplorerSortOrder64.dll", CallingConvention = CallingConvention.Cdecl, CharSet = CharSet.Unicode, EntryPoint = "GetExplorerSortOrder")]
+        public static extern int GetExplorerSortOrder64(string path, ref StringBuilder str, int len, ref Int32 ascend);
+
+        /// <summary>
+        /// Determines the sorting order of a Windows Explorer window which matches
+        /// the given file path.
+        ///
+        /// "Failure" situations are:
+        /// 1. unable to find an open Explorer window matching the file path
+        /// 2. the Explorer sort order doesn't match
+        /// </summary>
+        /// <param name="fullpath">full path to file/folder in question</param>
+        /// <param name="LoadOrder">the resulting sort order or null</param>
+        /// <param name="ascending">the resulting sort direction or null</param>
+        /// <returns>false on failure - out parameters will be null!</returns>
+        public static bool GetExplorerSortOrder(string fullpath, out ImageOrderBy? LoadOrder, out bool? ascending)
+        {
+            // assume failure
+            LoadOrder = null;
+            ascending = null;
+
+            try
+            {
+                string path = Path.GetDirectoryName(fullpath);
```

Comment:
Let's rename `path` to `dir` (or `dirPath`) because `Path.GetDirectoryName(fullpath)` returns directory path